### DISCUSSION TITLE
Apply black on source code

### DISCUSSION THREAD

**Discussion by Kludex** [celery/celery#7806] [Ideas]
What are the thoughts about it?

If the loss of commit history is an argument, we could use this: https://michaelheap.com/git-ignore-rev/.

**Comment by auvipy** [comment]
IMHO, it is not a big deal right now. we better focus on other issues now. will consider black in future as needed

**Reply by Kludex** [reply]
Well... Celery's source code is hard to understand. Not only because it is conceptually complex, but because it was created many years ago.

This would be one step forward to make the code source more understandable. If more people understand what is here, more people can help to improve this project.

**Reply by MelnykR** [reply]
As an alternative, [darker](https://pypi.org/project/darker) could be used.
It may be configured to apply black (and / or Mypy, pylint) only on changes made in particular PR. 
It's not perfect sometimes, but it helps to make codebase more readable. 

IMHO, the less code is understandable, the less people contribute and fix "big deal" issues.

**Reply by clokep** [reply]
+1 for switching to black. It provides a consistent style (which Celery is currently not) -- it really does help with finding bugs and making it easier for contributors to jump in.
